### ISSUE TITLE
VCs proof verification fails when multiple assertionKeys are present for a DID

### ISSUE THREAD

**Issue by stevenvegt** [nuts-foundation/nuts-node#2188]
Might be a false positive and must be researched. Encountered during the LSPxNuts hackathon on the VZVZ vendor document.

**Comment by woutslakhorst** [comment]
Code flow doesn't suggest anything of the sorts. Was there a specific error?

I added a 2nd key to my DID, issued a VC and got the following:
```
AUDIT[838101] Signing a JWS with key: did:nuts:HSr29Y6qFFSHGPzn9VK7dhpbT14U9epwPhMECRHBwdaQ#9FnwGYICLv1tmIMoAiD2wTIF7J1LHkRZVcudxDwAcDE  actor=demo@192.168.10.132 event=SignJWS module=Crypto operation=VCR.IssueVC
ERRO[838102] Notify event dropped                          error="invalid proof signature: failed to verify signature using ecdsa" eventSubscriber=vcr_vcs module=Network txRef=57adf1cd639da04e866fda43076df223d95cc2f7e3248fe4b5d497cd5b448ae2
INFO[838102] Transaction created                           module=Network txPayloadLen=1098 txRef=57adf1cd639da04e866fda43076df223d95cc2f7e3248fe4b5d497cd5b448ae2 txType=application/vc+json
INFO[838102] Verifiable Credential published               credentialID="did:nuts:HSr29Y6qFFSHGPzn9VK7dhpbT14U9epwPhMECRHBwdaQ#00875149-75c7-4d28-b875-5f785ac2f1bf" credentialType="[NutsOrganizationCredential VerifiableCredential]" module=VCR
```

**Comment by woutslakhorst** [comment]
When removing additional keys, VCs are accepted again. So it's the acceptance that fails.

**Comment by woutslakhorst** [comment]
Probably has something to do with key lookup

**Comment by woutslakhorst** [comment]
Problem is in `resolvers.go#ResolveRelationKey`

```
for _, rel := range relationships {
	if rel.ID.String() == keyID {
		result = &rel
	}
}
```
Spot the error ;)

**Comment by woutslakhorst** [comment]
closed by #2191
